src/actions/events.js

```diff
@@ -1,15 +1,74 @@
+import database from "../firebase/firebase";
+
+export const setEvents = events => ({
+  type: "SET_EVENTS",
+  events,
+});
+
+export const startSetEvents = () => (dispatch, getState) => {
+  const { uid } = getState().auth;
+
+  return database
+    .ref(`users/${uid}/events`)
+    .once("value")
+    .then(snapshot => {
+      const events = [];
+
+      snapshot.forEach(childSnapshot => {
+        events.push({
+          id: childSnapshot.key,
+          ...childSnapshot.val(),
+        });
+      });
+      dispatch(setEvents(events));
+    });
+};
+
 export const addEvent = event => ({
   type: "ADD_EVENT",
   event,
 });
 
+export const startAddEvent = eventData => (dispatch, getState) => {
+  const { uid } = getState().auth;
+
+  return database
+    .ref(`users/${uid}/events`)
+    .push(eventData)
+    .then(ref => {
+      dispatch(addEvent({ id: ref.key, ...eventData }));
```

Comment:
It's a good idea to place spread operator at the top of a new object. `rest` is an operator which should be at the end, but here you are using spread, not rest ;)